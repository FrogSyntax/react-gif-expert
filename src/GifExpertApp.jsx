import { useState } from "react";
import { AddCategory, GifGrid } from "./components";

export const GifExpertApp = () => {
    //! Es recomendable tener x cantidad de useState por cantidad de variables que necesitamos
    const [categories, setCategories] = useState(["One Punch", "Kimetsu"]);

    const deleteAllCategories = () => {
        setCategories([])
    }


    const onAddCategory = (newCategory) => {
        if (categories.map(x => x.toLowerCase()).includes(newCategory.toLowerCase())) return;
        // setCategories(categories.concat(valor));
        // setCategories([...categories, valor]);
        setCategories([ newCategory, ...categories]);
    };

    return (
        <>
            <h1>Gif Expert App</h1>
            <button onClick={deleteAllCategories}>Eliminar Categorias</button>
            <AddCategory 
                // setCategories={setCategories} 
                onNewCategory = {value => onAddCategory(value)}
            />
            {
                categories.map((category) => 
                    (
                        <GifGrid 
                            key={category} 
                            category={category}
                            setCategories={setCategories}
                        />
                        
                    )
                )
            }
        </>
    );
};
