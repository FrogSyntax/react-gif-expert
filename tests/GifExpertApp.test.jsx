import { fireEvent, render, screen } from "@testing-library/react"
import { GifExpertApp } from "../src/GifExpertApp"

describe('pruebas de gifExpertApp', () => {
    test('deberia eliminar las categorias', () => {
        render(<GifExpertApp/>);
        const [ButtonEliminarTodo, ] = screen.getAllByRole('button')
        fireEvent.click(ButtonEliminarTodo)
        expect(() => {screen.getAllByRole('heading', {level: 3})}).toThrow()
    })
    test('deberia eliminar la acategoria OnePunch', () => {
        render(<GifExpertApp/>);
        const [, ...ButtonEliminar] = screen.getAllByRole('button')
        // console.log(ButtonEliminar)
        fireEvent.click(ButtonEliminar[1])
        expect(screen.getAllByRole('heading', {level: 3}).length).toBe(1)
    })
    test('Deberia mostrar las Categorias iniciales', () => {
        render(<GifExpertApp/>);
        expect(screen.getAllByRole('heading', {level: 3}).length).toBe(2)
    })
    test('OnAdd category agrega una categoria al principio de la pagina', () => {
        const inputValue = 'Dragon Ball'
        render(<GifExpertApp/>);
        const input = screen.getByRole('textbox')
        const form = screen.getByRole('form')
        fireEvent.input(input,{target: {value: inputValue}})
        // screen.debug()
        fireEvent.submit(form)
        expect(screen.getByText(inputValue))
    })
    test('OnAdd category no agrega nada si le envian el mismo nombre', () => {
        const inputValue = 'KIMETSU'
        const {container} = render(<GifExpertApp/>);
        const input = screen.getByRole('textbox')
        const form = screen.getByRole('form')
        fireEvent.input(input,{target: {value: inputValue}})
        fireEvent.submit(form)
        screen.debug()
        
        expect(container).toMatchSnapshot()
    })
})