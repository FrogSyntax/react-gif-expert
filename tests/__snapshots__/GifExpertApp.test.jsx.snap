// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas de gifExpertApp OnAdd category no agrega nada si le envian el mismo nombre 1`] = `
<div>
  <h1>
    Gif Expert App
  </h1>
  <button>
    Eliminar Categorias
  </button>
  <form
    aria-label="form"
  >
    <input
      placeholder="Buscar gifs"
      type="text"
      value=""
    />
  </form>
  <h3>
    One Punch
  </h3>
  <button>
    Elminar Categoria
  </button>
  <h2>
    Cargando...
  </h2>
  <div
    class="card-grid"
  />
  <h3>
    Kimetsu
  </h3>
  <button>
    Elminar Categoria
  </button>
  <h2>
    Cargando...
  </h2>
  <div
    class="card-grid"
  />
</div>
`;
